src/main/scala/com/touk/hiring_task/multiplex/repositories/CinemaRepository.scala

```diff
@@ -0,0 +1,60 @@
+package com.touk.hiring_task.multiplex.repositories
+
+import com.touk.hiring_task.multiplex.commons.{MultiplexException, SelectedPlaceUnavailableException}
+import com.touk.hiring_task.multiplex.dto.OrderDto
+import com.touk.hiring_task.multiplex.model._
+import com.touk.hiring_task.multiplex.utils.SystemClock
+
+import java.util
+import java.util.Enumeration
+import java.util.concurrent.ConcurrentHashMap
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CinemaRepository {
+  def fetchProjectionsByDate(startAt: Long, finishAt: Long): Future[Option[List[Projection]]]
+
+  def takeProjectionById(id: String): Future[Option[Projection]]
+
+  def fetchKeysDB(): Future[Option[Enumeration[Key]]]
+
+  def storeReservations(o: OrderDto, expire: Long, roomName: String): Future[Either[MultiplexException, Unit]]
+}
+
+class CinemaRepositoryImpl(fillDB: FillCinemaRoomsService)(implicit ec: ExecutionContext) extends CinemaRepository with SystemClock {
+
+  val projectionsList = fillDB.getListProjections()
+  val cinemaDB = fillDB.getCinemaRooms(projectionsList)
+
+  override def fetchProjectionsByDate(projectionStartAt: Long, projectionFinishAt: Long): Future[Option[List[Projection]]] =
```

Comment:
What does it mean when the method returns `Future.successful(Some(Nil))`? If it's just an invalid state, do you know a type that helps us to make the invalid state impossible in compile time?